proto/validatorstate/validator_state.proto

```diff
@@ -58,6 +74,27 @@ message Validator {
   bool is_l1_validator = 8; // can be empty
 }
 
+message GetWarpValidatorSetsResponse {
+  repeated WarpValidatorSet validator_sets = 1;
+}
+
+message GetWarpValidatorSetResponse {
+  uint64 total_weight = 1;
+  repeated WarpValidator validators = 2;
+}
+
+message WarpValidatorSet {
+  bytes subnet_id = 1;
+  uint64 total_weight = 2;
+  repeated WarpValidator validators = 3;
+}
+
+message WarpValidator {
+  bytes public_key = 1; // Uncompressed public key, must not be empty
+  uint64 weight = 2;
+  repeated bytes node_ids = 3; // must not be empty
```

Comment:
[No action] Relayers/Aggregators will need the `node_ids` to know which nodes to query for signatures, but I don't think VMs actually need them to verify an ICM signature.

I'm guessing that this needs to support both use cases, and isn't worth over optimizing for the VM case?

Reply:
Yeah the only usage of `NodeIDs` is with the aggregator code. I don't think it will really matter because this evaluation will be cached.